lib/network_utils.pm

```diff
@@ -90,8 +90,13 @@ sub recover_network {
 }
 
 sub ifc_exists {
-    my ($ifc) = @_;
-    return !script_run('ip link show dev ' . $ifc);
+    my ($ifc, $check_is_up) = @_;
```

Comment:
WDYT, Instead of adding a flag which controls the flow of that function, add a new function `ifc_is_up()` which internally calls `ifc_exists()` + that extra check.

Reply:
don't like one liner functions , as for me they introduce more problem than solves , it is very annoying to jump over code here and there just to see how exactly ping is done or something like this

Reply:
also I didn't get how wrapping this into separate function will change the fact that we need to control whether we want to do this check or not?

Reply:
For me, the benefit is in reading the code.

Reply:
I think `ifc_is_up('eth0')` is more readable then `ifc_exists('eth0', 1)`.

Reply:
> I think `ifc_is_up('eth0')` is more readable then `ifc_exists('eth0', 1)`.

I still didn't get this part , if @jlausuch will implement your idea how `ifc_exists` call will look like ? 
`ifc_exists('eth0')` ?

Reply:
E.g. the line https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7609/files#diff-7c6669e0bb0ed1f0eed4a62dc81f200dR514 would look like 

```
  if (!ifc_is_up($interface)) {
```

Reply:
yes, see new patchset